File: digest-util/src/test/java/cc/duduhuo/util/digest/TestHex.java
/*
 * Copyright 2017-present Li Ying.
 * Licensed under the MIT License.
 */

package cc.duduhuo.util.digest;

import org.junit.jupiter.api.Test;

import java.io.File;
import java.io.IOException;

import static cc.duduhuo.util.digest.Hex.hex;
import static org.junit.jupiter.api.Assertions.assertEquals;


public class TestHex {
    @Test
    public void test() throws IOException {
        assertEquals("616263", hex("abc".getBytes()));
        String s = hex(new File("for-testing-only.txt"));
        System.out.println("hex = " + s);
        assertEquals(998, s.length());
    }
}

File: digest-util-ktx/src/test/kotlin/cc/duduhuo/util/digest/ktx/TestHex.kt
/*
 * Copyright 2017-present Li Ying.
 * Licensed under the MIT License.
 */

package cc.duduhuo.util.digest.ktx

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.io.File
import java.io.IOException

class TestHex {
    @Test
    @Throws(IOException::class)
    fun test() {
        Assertions.assertEquals("616263", "abc".toByteArray().hex())
        val s = File("for-testing-only.txt").hex()
        println("hex = $s")
        Assertions.assertEquals(998, s.length)
    }
}
